src/api/api.service.ts

```diff
@@ -32,89 +37,162 @@ export class ApiService {
+          ),
+        );
+      } else {
+        const status = api.profileId === profileId ? true : false;
+        return status;
+      }
+    } catch (err) {
+      throw new BadRequestException(
+        ZaLaResponse.BadRequest('Internal Server error', err.message, '500'),
+      );
+    }
+  }
+
+  async deleteApi(apiId: string, profileId: string) {
+    try {
+      const apiExist = await this.apiRepo.findOne({ where: { id: apiId } });
+      const isOwner = await this.verify(apiId, profileId);
+
+      if (isOwner === false) {
+        throw new NotFoundException(
+          ZaLaResponse.BadRequest('Forbidden', 'Unauthorized action', '403'),
+        );
+      }
+
+      if (apiExist && isOwner === true) {
+        return await this.apiRepo.remove(apiExist);
+      }
+      throw new NotFoundException(
+        ZaLaResponse.NotFoundRequest('Not Found', 'Api does not exist', '404'),
+      );
+    } catch (err) {
+      throw new BadRequestException(
+        ZaLaResponse.BadRequest('Internal Server error', err.message, '500'),
+      );
     }
+  }
+
+  /**
+   * It checks if the user is the owner of the api, if yes, it updates the api.
+   * </code>
+   * @param {string} apiId - The id of the api to be updated.
+   * @param {string} profileId - The id of the user who is trying to update the api.
+   * @param {UpdateApiDto} updateApiDto - UpdateApiDto
+   * @returns The update method returns a promise of type UpdateResult.
+   */
 
-    async verify(apiId: string, profileId: string) {
-        try {
-            const api = await this.apiRepo.findOne({ where: { id: apiId }});
-            if(!api) {
-                throw new NotFoundException(
-                    ZaLaResponse.NotFoundRequest(
-                        'Not found',
-                        'The api does not exist',
-                        '404',
-                    ),
-                );
-            } else {
-                const status = api.profileId === profileId ? true: false;
-                return status;
-            }
+  async update(
+    apiId: string,
+    profileId: string,
+    updateApiDto: UpdateApiDto,
+  ): Promise<Api> {
+    try {
+      const api = await this.apiRepo.findOne({ where: { id: apiId } });
+      if (api) {
+        /* Checking if the user is the owner of the api. */
+        const verified = await this.verify(apiId, profileId);
+        if (verified === false) {
+          throw new NotFoundException(
+            ZaLaResponse.BadRequest('Forbidden', 'Unauthorized action', '403'),
+          );
         }
-        catch(err) {
-            throw new BadRequestException(
-                ZaLaResponse.BadRequest('Internal Server error', err.message, '500'),
-            );
+
+        /* Checking if categoryID exist in the api. */
+        const findCategory = await this.apiRepo.findOne({
+          where: { categoryId: updateApiDto.categoryId },
+        });
+
+        if (!findCategory) {
+          throw new NotFoundException(
```

Comment:
What will be the effect of this block of code assuming the user isn't updating the category? This should only run if the user is updating category too